packages/esm-patient-chart-app/src/encounter-list/components/encounter-list.component.tsx

```diff
@@ -0,0 +1,323 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { navigate, showModal, showSnackbar } from '@openmrs/esm-framework';
+import { EmptyState } from '@openmrs/esm-patient-common-lib';
+import { useTranslation } from 'react-i18next';
+import { EncounterListDataTable } from './table.component';
+import {
+  Button,
+  Link,
+  OverflowMenu,
+  OverflowMenuItem,
+  DataTableSkeleton,
+  MenuButton,
+  MenuItem,
+  Pagination,
+} from '@carbon/react';
+import { Add } from '@carbon/react/icons';
+import { launchEncounterForm } from '../../clinical-views/utils/helpers';
+import { deleteEncounter } from '../encounter-list.resource';
+import { useEncounterRows, useFormsJson, usePatientDeathStatus } from '../hooks';
+
+import styles from './encounter-list.scss';
+import { type TableRow, type Encounter, type Mode } from '../types';
+
+export interface EncounterListColumn {
+  key: string;
+  header: string;
+  getValue: (encounter: any) => string;
+  link?: any;
+}
+
+export interface EncounterListProps {
+  patientUuid: string;
+  encounterType: string;
+  columns: Array<any>;
```

Comment:
This should be explicitly typed.

Reply:
In the `EncounterListTabsComponent`, this gets the type `FormattedColumn`, couldn't we use the same here, instead of any.